src/IB/IBFEMethod.cpp

```diff
@@ -1017,6 +1022,41 @@ IBFEMethod::spreadForce(const int f_data_idx,
         d_ghost_data_accumulator->accumulateGhostData(f_scratch_data_idx);
     }
 
+    // Prolong forces spread onto coarser levels onto finer levels.
+    if (d_use_scratch_hierarchy)
+    {
+        // not yet implemented
+        assertStructureOnFinestLevel();
+    }
+    else
+    {
+        // TODO - we should save these schedules in the same manner that we save
+        // the schedules for the scratch hierarchy.
+        for (int ln = getCoarsestPatchLevelNumber(); ln < getFinestPatchLevelNumber(); ++ln)
+        {
+            Pointer<RefineAlgorithm<NDIM> > refine_algorithm = new RefineAlgorithm<NDIM>();
+            Pointer<CartesianGridGeometry<NDIM> > geometry = d_hierarchy->getGridGeometry();
+            // TODO: this is probably the wrong refinement operator
+            Pointer<RefineOperator<NDIM> > refine_op =
```

Comment:
```suggestion
            Pointer<RefineOperator<NDIM> > refine_op = d_f_prolong_op;
```

Reply:
I'm stuck on this one - I am still looking for a way to pull this out of `IBExplicitHierarchyIntegrator`. As far as I can tell `RefineOperator`s get assigned to `RefineAlgorithm`s and then cannot be retrieved again.

Reply:
If you have a `RefineSchedule`, you can eventually get to the `RefineOperator` through `RefineSchedule::getEquivalenceClasses()`.

Reply:
You can do the same with `RefineAlgorithm::getEquivalenceClasses()`.

Reply:
But it might be better to add something to the `IBHierarchyIntegrator` interface to make it easier to access.

The model that I had in mind when setting up `IBHierarchyIntegrator` was that `IBStrategy` implementations would only need the `RefineSchedule`s and not need to know any details.

Reply:
Could you use the existing interface if you first made a copy of the level-by-level force data and then use the `RefineSchedule`s passed in via `IBStrategy::spreadForce()`?

If you don't like the idea of copying data, SAMRAI also makes it easy to "swap" storage associated with two patch data indices.

Reply:
BTW, looking at `IBHierarchyIntegrator`, the prolongation and refine operators are hard coded anyway. Probably they should be made to be run-time options.